tests/test_weaver_snapshot_locking.py

```
@@ -73,6 +74,47 @@ def test_the_startup_lock_is_released_when_the_run_fails(
+    tmp_path: Path, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """Only contention is worth waiting out.
+
+    A kernel with no lock table left (ENOLCK) will not grow one because
+    another run finished, so retrying turns a hard failure into a
+    thirty-second stall that ends in the wrong message.
+    """
+    lock = tmp_path / "port.lock"
+    attempts: list[int] = []
+
+    def _no_locks_left(handle: object, flags: int) -> None:
```

Ruff (0.16.2)

[warning] Unused function argument: `handle`

(ARG001)